### ISSUE TITLE
Properties with "base" in the name show wrong content or cannot be expanded

### ISSUE THREAD

**Issue by Lexikos** [zero-plusplus/vscode-autohotkey-debug#163]
```AutoHotkey
test1 := {baseType: 10}
test2 := {}, test2.DefineProp('base', {value: 10})
test3 := {baseType: {N: 'Test3'}}
test4 := {}, test4.DefineProp('base', {value: {N: 'Test4'}})
test5 := {unbased: {N: 'Test5'}}
MsgBox
```
When the extension queries the content of these objects, it incorrectly replaces `base` with `<base>`.
  - The values returned by `context_get` when the variable list is initially populated appear correct.
  - `test2` shows as `Object {}`; i.e. the user-defined `base` property is omitted. That's avoidable, but okay.
  - The value of `test3.baseType.N` is not shown when expanding, because the extension is querying for the invalid property `test3.<base>Type.N`.
  - The second `test4.base` is initially shown correctly as `Object {__Class: "Object"...`, but when expanded, it shows the content of the true base object, not the user-defined property. This is because the extension is querying for `test4.<base>` even though that is not the name the engine originally returned.
  - `test5.unbased` correctly shows as `Object {N: "Test5"}`, because apparently the extension is correctly querying for `test5.unbased`, not `test5.un<base>d`.
  - `test5.unbased.base` cannot be expanded, because the extension is querying for `test5.un<base>d.<base>`.

The engine represents the true base as `<base>` so that if the user defines another property by that name (or in v1, "base" is used as a key in an associative array), the debugger client can still display the true base. When this was originally added (in v1.1.30), the existing clients just worked; they received a property like `test2.<base>`, displayed it as is, and queried it as is, returning the correct object. There was no need for any transformation between `<base>` and `base` on the client side. The only drawback was that the user interface showed `<base>`, which is a bit odd, although it might be less confusing than having two properties displayed as `base`.

I suggest you retain the actual property name returned by the engine, and always query using that exact name. If you can do that while displaying `base` in the UI, I leave that up to you. It's probably better to display `base` when there's no ambiguity.

**Comment by Lexikos** [comment]
Actually, another purpose of `<base>` is to query the base properties while keeping `this` as the context; i.e. if there's a property getter, `this.base.x` will call it with `this.base` as the first parameter while `this.<base>.x` will call it with `this`.

**Comment by zero-plusplus** [comment]
I've checked.

I started using base instead of `<base>` when I noticed in [demo.ahk](https://github.com/zero-plusplus/vscode-autohotkey-debug/blob/master/demo/demo.ahk) that I could not get any value when I checked `instance.<base>.<base>`.
This did not happen when I used a watch expression to get the value like `instance.base.base`, and I used `base` as a workaround.

The issue I described above was the same for [helsmy's DebugAdapter](https://marketplace.visualstudio.com/items?itemName=helsmy.autohotkey-debug).

I didn't investigate too deeply because it worked fine with this workaround.

---

This issue also needs to be considered together with #50.

**Comment by zero-plusplus** [comment]
It is not appropriate to show `<base>` in autocomplete, so I need to consider that as well.

---

In the issue above, `<base>`, which is supposed to be there, returns undefined, so I solved it by using `base` to re-fetch it only in that case.

**Comment by Lexikos** [comment]
> Following this, autocomplete will now suggest base if the property is user-defined base, and <base> if it is not.

Why?  `<base>` should never be offered as an option, because it is only for the debugger.

> ... the property that is user-defined base will be retrieved first.

`obj.base` always returns the *defined property*, period. The built-in base property is not a special case, just a normal property defined in `Any.prototype`.

> In the issue above, `<base>`, which is supposed to be there, returns undefined

This was a bug, fixed by Lexikos/AutoHotkey_L@f65d726 for v2.0-a104 and Lexikos/AutoHotkey_L@0c479d0 for v1.1 (unreleased).

Actually, I suppose it applied to `x.y.<base>` as well.

**Comment by zero-plusplus** [comment]
> Why? <base> should never be offered as an option, because it is only for the debugger.

You are right, it only displays `base`.

> obj.base always returns the defined property, period. The built-in base property is not a special case, just a normal property defined in Any.prototype.

Thank you. I now understand how it works.

> In the issue above, <base>, which is supposed to be there, returns undefined

Thanks for the fix. I'll keep the workaround to support current and past versions, and change to a simpler implementation after the fix.
